File: app/src/main/java/com/mnj/java_solutions/TwoSumSortedArray.java
package com.mnj.java_solutions;

public class TwoSumSortedArray {

    void twoSumPair() {

        int[] num = {2,3,4};

        int target = 6;

        int i = 0;
        int j = num.length - 1;

        while (i < j) {
            if ((num[i] + num[j]) == target) {
                System.out.println("==>> pair is :" + i + " , " + j);
                i++;
                j--;
            } else if ((num[i] + num[j]) < target) {
                i++;
            } else {
                j--;
            }
        }
    }
}

File: app/src/main/java/com/mnj/java_solutions/twopointer/TwoSumSortedArray.kt
package com.mnj.java_solutions.twopointer

class TwoSumSortedArray {
    fun twoSumPair() {
        val num = intArrayOf(2, 3, 4)
        val target = 6
        var i = 0
        var j = num.size - 1
        while (i < j) {
            if (num[i] + num[j] == target) {
                println("==>> pair is :${num[i]} , ${num[j]}")
                i++
                j--
            } else if (num[i] + num[j] < target) {
                i++
            } else {
                j--
            }
        }
    }
}